compiler/rustc_middle/src/ty/error.rs

```diff
@@ -309,11 +309,12 @@ impl<'tcx> Ty<'tcx> {
 
 impl<'tcx> TyCtxt<'tcx> {
     pub fn ty_string_with_limit(self, ty: Ty<'tcx>, length_limit: usize) -> String {
-        let mut type_limit = 50;
         let regular = FmtPrinter::new(self, hir::def::Namespace::TypeNS)
             .pretty_print_type(ty)
             .expect("could not write to `String`")
             .into_buffer();
+
+        let mut type_limit = 50;
```

Comment:
I don't think this change is paying for itself. (I.e. there's a cost with changing code unnecessarily, in terms of impeding people who e.g. use `git annotate` as part of their workflow.) So I'd revert this back to the prior form.